sigstore/_internal/rekor/client.py

```diff
@@ -246,7 +247,10 @@ def log(self) -> RekorLog:
         """
         Returns a `RekorLog` adapter for making requests to a Rekor log.
         """
-        return RekorLog(f"{self.url}/log", session=self.session)
+
+        # Each RekorLog gets their own session
+        with self._session() as s:
+            return RekorLog(f"{self.url}/log", session=s)
```

Comment:
I _think_ we want this, since a `with ...` block will call `__del__` on `s` on cleanup (which then escapes via the `return`). But I might be under-thinking that.

```suggestion
        return RekorLog(f"{self.url}/log", session=self._session())
```

Reply:
:+1: yeah, at the very least my `with` does nothing useful here